homeassistant/components/lastfm/sensor.py

```diff
@@ -84,17 +84,25 @@ def update(self):
         """Update device state."""
         self._cover = self._user.get_image()
         self._playcount = self._user.get_playcount()
-        last = self._user.get_recent_tracks(limit=2)[0]
-        self._lastplayed = f"{last.track.artist} - {last.track.title}"
-        top = self._user.get_top_tracks(limit=1)[0]
-        toptitle = re.search("', '(.+?)',", str(top))
-        topartist = re.search("'(.+?)',", str(top))
-        self._topplayed = f"{topartist.group(1)} - {toptitle.group(1)}"
-        if self._user.get_now_playing() is None:
+
+        recent_tracks = self._user.get_recent_tracks(limit=2)
+        if len(recent_tracks) != 0:
```

Comment:
You can use the variable as a truly value, only passing the condition if it is not empty.

```suggestion
        if recent_tracks:
```